src/main/java/com/aws/greengrass/secretmanager/SecretManagerIPCAgent.java

```diff
@@ -5,32 +5,124 @@
+    public SecretManagerIPCAgent(SecretManager secretManager, AuthorizationHandler authorizationHandler) {
+        this.secretManager = secretManager;
+        this.authorizationHandler = authorizationHandler;
+    }
 
     public GetSecretValueOperationHandler getSecretValueOperationHandler(OperationContinuationHandlerContext context) {
         return new GetSecretValueOperationHandler(context);
     }
 
+    /**
+     * Handles secret API calls from lambda-manager for v1 style lambdas.
+     *
+     * @param serviceName Lambda component name requesting the secret.
+     * @param request     v1 style get secret request
+     * @return v1 style secret or error serialized as JSON bytes. If a serialization error occurs, null is returned.
+     */
+    public byte[] getSecret(String serviceName, byte[] request) {
+        logger.atInfo().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
+                .log("requested secret");
+
+        int status;
+        String message = null;
+
+        try {
+            com.aws.greengrass.secretmanager.model.v1.GetSecretValueRequest getSecretValueRequest =
+                    OBJECT_MAPPER.readValue(request,
+                            com.aws.greengrass.secretmanager.model.v1.GetSecretValueRequest.class);
+            validateSecretIdAndDoAuthorization(GET_SECRET_VALUE, serviceName, getSecretValueRequest.getSecretId());
+            GetSecretValueResult response = secretManager.getSecret(getSecretValueRequest);
+
+            try {
+                return OBJECT_MAPPER.writeValueAsBytes(response);
+            } catch (JsonProcessingException e) {
+                // not logging exception in case it outputs secret value
+                logger.atError().event("secret-access").kv("Principal", serviceName).kv("secret", new String(request))
+                        .log("Error serializing secret");
+                return null;
+            }
+        } catch (IOException e) {
+            status = 400;
+            message = "Unable to parse request";
+        } catch (AuthorizationException t) {
+            status = 403;
+            message = t.getMessage();
+        } catch (GetSecretException t) {
+            status = t.getStatus();
+            message = t.getMessage();
+        } catch (Throwable t) {
+            status = 500;
+            message = t.getMessage();
+            logger.atError().event("secret-access").setCause(t).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).log("Error getting secret");
+        }
+
+        try {
+            return OBJECT_MAPPER.writeValueAsBytes(
+                    GetSecretValueError.builder().status(status).message(message).build());
+        } catch (JsonProcessingException e) {
+            logger.atError().event("secret-access").setCause(e).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).kv("status", status).kv("message", message)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```